docs/design-proposals/cloud-edge-communication/cloud-edge-communication-framework.md

```diff
@@ -30,28 +30,67 @@ edge server -----> Proxy A(in the edge) <----------> Proxy B(in the cloud) -----
+
+If the proxy want to support message(rabbitmq/kafka) between the cloud and the edge. The proxy works as a amqp/kafka server, it can proxy the message call/connection to the destination site, it need to understand the amqp/kafka protocol, to do the connection and request forward.
+And also there are encryption/decryption problems(MitM). It is hard to support all the communcation protocols in this way.
+
+So if the rest call(http/https request) is the only communication type between the cloud and the edge, maybe it is a simple/good choice. 
+
+Option 2: The proxy works at network layer 4, do the udp/tcp proxy.
+
+For example, we want to support rest calls(http/https), message queue connection(rabbitmq/kafka), and something else based on the udp protocol.
+The proxy works as a udp server, and it forward the udp connection/traffic to the destination site.
+
+It works just like a nginx, but there are some differences:
+1. It is not proxy the traffic to the destination service directly, but to the other proxy in other site
+2. It does not work only for one or two services, and with static configuration. It needs to support multiple services at the same time, and the services can be dynamic add or remove.
+It needs to solve the port allocation and traffic encryption/decryption problems
+
+![](Proposal-B4.png) 
 
 ## Proposal B
 
-We want to implement it based on network layer 3, and we also need to solve the ip address conflict and security problems.
+We want to implement the communication framework based on the network layer 3, and also we need to solve the ip address conflict and security problems. We can use ip translation to do this. 
+
+We need to add the gateway/proxy for the edge site and the cloud for the communication, we call it edgegateway in each edge site, and in addition to communication, cloudgateway in responsible for central control. We don't expose the network to the cloud or the edge site(cause too much attach surface), we expose the services to the gateway.  
 
-We want to introduce a new concept virtual presence:
+Here we want to introduce a concept "virtual presence":
 * It is a local virtual presentation of remote services
 * It is full isolation of remote servers
-* It exposes services to remote sites, do not expose networks
-* It uses locally assigned virtual presence ips to avoid exposure of site internal addresses
-* It use translation mechanisms to emulate local traffic flow
-* There is no man-in-the-middle, no session termination, it provide end-to-end encryption
 
-![](Proposal-B1.png)
+If the service is exposed to the edge site, there will be a virtual presence for the service in this edge site. The clients in this edge site which want to communicate to the service will be told to communicate to the virtual presence of the service which is on the gateway of the edge site. And our gateway will forward traffic in secured tunnel using local virtual presence ip address to the destination gateway and services.
```

Comment:
just so I understand correctly: the difference between this solution and the current VPN solution is that clients on the edge site do not communicate with the service in the cloud using their actual ip, but the ip of the gateway instead. gateway still talks to the cloud (the cloudgateway actually) through VPN. This reduces the attack surface to just the edgegateway.

Reply:
"gateway still talks to the cloud (the cloudgateway actually) through VPN. "    gateway can talk to the cloud through VPN, or with security tunnel which is build by the gateways.  Currently we choose the second option, we don't want to use the VPN for our cloud-edge communication.